### PR TITLE
Add support for Ruby 2.3.1 which is used in the new Puppet 4 AIO

### PR DESCRIPTION
**Pull request by ghoneycutt**

### CODE REVIEW

**Comment by ghoneycutt** [comment]
Tests are passing now. @jenkinsci @rtyler could you please have a look

**Comment by ghoneycutt** [comment]
Well one of the beaker tests failed, though it is unrelated to this.

**Comment by ghoneycutt** [comment]
Hi, could someone please have a look

**Comment by ghoneycutt** [comment]
Hello?

**Comment by jhoblitt** [comment]
I've kicked travis.  I'm not sure why beaker failed.

File: Gemfile [hunk 1]
```diff
@@ -13,6 +13,8 @@ gem 'retries', '~> 0.0.5'
 gem 'travis', '~> 1.8'
 gem 'parallel_tests', :platform => [:mri_20, :mri_21]
 gem 'rubocop', '< 0.42.0'
+gem 'public_suffix', '1.4.6', :require => false if RUBY_VERSION <= '1.9.3'
+gem 'public_suffix',          :require => false if RUBY_VERSION > '1.9.3'
```
Comment (jhoblitt) [inline Gemfile]: We've dropped ruby < 2 from the travis matrix completely.  Could you remove this and rebase on current master?

**Comment by jhoblitt** [comment]
@ghoneycutt I was going to rebase and merge this but it looks like `puppet-agent-1.8.2-1.el7.x86_64` is still using ruby 2.1.  What new AIO version are you referring to?

```
[root@centos-7-x64 bin]# ./puppet --version
4.8.1
[root@centos-7-x64 bin]# ./ruby --version     
ruby 2.1.9p490 (2016-03-30 revision 54437) [x86_64-linux]
```

**Comment by jhoblitt** [comment]
Ruby 2.4 was added in #790 and the acceptance tests use the AIO packaging.  I'm going to go ahead and close this issue but please re-open it if there is a reason to add 2.3 to the travis matrix.

**Comment by ghoneycutt** [comment]
2.3.1 never made it in. Puppet v4.5.0 - v4.10 use ruby 2.1.9 and Puppet 5 is using 2.4.1

**Comment by ghoneycutt** [comment]
Here's the travis config I'm using - https://github.com/sensu/sensu-puppet/blob/master/.travis.yml

**Pull request closed without merging**